app/(admin)/admin/leagues/page.tsx

```diff
@@ -1,34 +1,48 @@
 // Copyright (c) Gridiron Survivor.
 // Licensed under the MIT License.
 
-import { JSX } from 'react';
-import { LeagueCard } from '@/components/LeagueCard/LeagueCard';
+'use client';
+import { getAllLeagueEntries } from '@/api/apiFunctions';
+import { getUserLeagues } from '@/utils/utils';
+import { IEntryWithLeague } from '@/components/TableColumns/TableColumns.interface';
+import { JSX, useEffect, useState } from 'react';
+import { leagueColumns } from '@/components/TableColumns/TableColumns';
+import TableData from '@/components/TableData/TableData';
+import { useDataStore } from '@/store/dataStore';
 
 /**
  * Renders the admin page.
  * @returns {JSX.Element} - The rendered Admin Leagues page.
  */
 const AdminLeagues = (): JSX.Element => {
+  const [leaguesData, setLeaguesData] = useState<IEntryWithLeague[]>([]);
+  const { user } = useDataStore((state) => state);
+
+  /**
+   * Get all leagues the user is a part of.
+   */
+  const fetchData = async (): Promise<void> => {
+    const entries = await getAllLeagueEntries({ leagues: user.leagues });
+    const leagues = await getUserLeagues(user.leagues);
+    const combinedData = leagues.map((league, index) => ({
+      aliveEntries: entries[index].alive,
+      leagueId: '',
+      leagueName: league.leagueName,
+      logo: '',
+      participants: league.participants,
+      survivors: league.survivors,
+      totalEntries: entries[index].totalEntries,
+    }));
+    setLeaguesData(combinedData);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, [user.leagues]);
```

Comment:
Q: Does `fetchData()` require `useEffect()` for this to work properly?

Reply:
it has to be called somehow when the user get's to the page. i used the useEffect() so it calls every time the user's league information gets updated (which is the live data being shown in the table). I may delete [user.leagues] so it only gets called once on page load